pkg/yamlmeta/node.go

```diff
@@ -69,13 +70,26 @@ func (a *Array) SetValue(val interface{}) error {
 }
 
 func (ai *ArrayItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into arrayitem", val)
+	if !(val == nil || isMapOrArrayOrScalar(val)) {
+		return fmt.Errorf("arrays can contain only maps, arrays, or scalars; this is a %T", val)
 	}
 	ai.Value = val
 	return nil
 }
 
+func isMapOrArrayOrScalar(val interface{}) bool {
+	switch val.(type) {
+	case *Map, *orderedmap.Map:
+		return true
+	case *Array, []interface{}:
+		return true
+	case int, int8, int16, int32, int64, bool, string, float32, float64:
```

Comment:
uints? not sure if that matters.

Reply:
Yes. The point is to permit _all_ legit values. Thank you!

Reply:
yup, i ended up looking at go docs to understand what other types are available. this also just reminded me of this: https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/template/core/go_value.go#L39

it would probably be good to also panic for unordered maps, similar to this: thttps://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/template/core/go_value.go#L39

Reply:
>yup, i ended up looking at go docs to understand what other types are available. this also just reminded me of this: https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/template/core/go_value.go#L39

🤔 

👉🏼  `yamlmeta` viewed as a mini-library will exhaustively allow-list values onto nodes.
👉🏼  `template.core` in so much that Starlark is an intrinsic integration can safely rely on the specific types emitted through the Go port of the language.

Reply:
> it would probably be good to also panic for unordered maps, similar to this: https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/template/core/go_value.go#L39

One advantage of returning an `error` over `panic`'ing is being able to report more context... it's the symptom that drove this PR in the first place. It helps answer the question, "from where did that raw Go map come from?"